### PR TITLE
Update the diff places bad opcode template to account for forks, and accounts for the changes in Cancun

### PR DESCRIPTION
**Pull request by qbzzt**
Closing: #1298 

Should be merged before merging #1297.

### CODE REVIEW

**Comment by winsvega** [comment]
can keep only src files

File: src/GeneralStateTestsFiller/stBadOpcode/invalidDiffPlacesFiller.yml [hunk 1]
```diff
@@ -35,7 +35,7 @@ invalidDiffPlaces:
            invalid()
             
             // If we get here, kill the goat so we'll get an error
-            mstore(0, hex"DEAD60A7")
+            mstore(0,  hex"DEAD60A7")
```
Comment (winsvega) [inline src/GeneralStateTestsFiller/stBadOpcode/invalidDiffPlacesFiller.yml]: only white space here, maybe can fix it?

File: src/Templates/DiffPlaces/createAllBadOpcodeTests.sh [hunk 2]
```diff
@@ -28,18 +26,17 @@
 ./createBadOpcodeTest.sh 2D
 ./createBadOpcodeTest.sh 2E
 ./createBadOpcodeTest.sh 2F
-./createBadOpcodeTest.sh 49
-./createBadOpcodeTest.sh 4A
+./createBadOpcodeTest.sh 49 '<Cancun'
```
Comment (winsvega) [inline src/Templates/DiffPlaces/createAllBadOpcodeTests.sh]: can it accept list?  `Berlin, London, Merge, Shanghai` ?

File: src/Templates/DiffPlaces/createAllBadOpcodeTests.sh [hunk 3]
```diff
@@ -28,18 +26,17 @@
...
 ./createBadOpcodeTest.sh 4C
 ./createBadOpcodeTest.sh 4D
 ./createBadOpcodeTest.sh 4E
 ./createBadOpcodeTest.sh 4F
-./createBadOpcodeTest.sh 5C
-./createBadOpcodeTest.sh 5D
-./createBadOpcodeTest.sh 5E
-# As of Shanghai 5F is no longer an invalid opcode
-# ./createBadOpcodeTest.sh 5F
+./createBadOpcodeTest.sh 5C '<Cancun'
```
Comment (winsvega) [inline src/Templates/DiffPlaces/createAllBadOpcodeTests.sh]: would be nice to have a comment here describing what is 5c opcode name

File: src/Templates/DiffPlaces/createAllBadOpcodeTests.sh [hunk 4]
```diff
@@ -28,18 +26,17 @@
...
 ./createBadOpcodeTest.sh 2F
-./createBadOpcodeTest.sh 49
-./createBadOpcodeTest.sh 4A
+./createBadOpcodeTest.sh 49 '<Cancun'
+./createBadOpcodeTest.sh 4A '<Cancun'
 ./createBadOpcodeTest.sh 4B
 ./createBadOpcodeTest.sh 4C
 ./createBadOpcodeTest.sh 4D
 ./createBadOpcodeTest.sh 4E
 ./createBadOpcodeTest.sh 4F
```
Comment (winsvega) [inline src/Templates/DiffPlaces/createAllBadOpcodeTests.sh:35]: it is still implicit that the script generates only >=London by default

File: src/Templates/DiffPlaces/createAllBadOpcodeTests.sh [hunk 5]
```diff
@@ -103,28 +100,28 @@
...
-./createBadOpcodeTest.sh E5
-./createBadOpcodeTest.sh E6
-./createBadOpcodeTest.sh E7
-./createBadOpcodeTest.sh E8
-./createBadOpcodeTest.sh E9
-./createBadOpcodeTest.sh EA
-./createBadOpcodeTest.sh EB
-./createBadOpcodeTest.sh EC
-./createBadOpcodeTest.sh ED
+./createBadOpcodeTest.sh E0 '<Cancun'
```
Comment (winsvega) [inline src/Templates/DiffPlaces/createAllBadOpcodeTests.sh]: E0 - ED  on cancun are not bad?

File: src/Templates/DiffPlaces/templateGen.js [hunk 6]
```diff
@@ -679,7 +681,7 @@ ${testName}:
         value: !!int -1
 
       network:
-        - '>=London'
+        - '${fork}'
```
Comment (winsvega) [inline src/Templates/DiffPlaces/templateGen.js]: if yaml can fit here fork into list  
```
- Berlin
- Merge
- Shanghai
```

would be nice

**Review by winsvega — COMMENTED** [review]
cosmetic changes wellcome

File: src/Templates/DiffPlaces/createAllBadOpcodeTests.sh (hunk 2 above)
Comment (qbzzt) [reply src/Templates/DiffPlaces/createAllBadOpcodeTests.sh]: Not yet. I can add that, when I have time - but I can't promise it'll be this week. Sorry, we are revamping our docs site.

File: src/Templates/DiffPlaces/createAllBadOpcodeTests.sh (hunk 4 above)
Comment (qbzzt) [reply src/Templates/DiffPlaces/createAllBadOpcodeTests.sh:35]: Yes, that's still the default. Should I change it to `>=Shanghai`?

File: src/Templates/DiffPlaces/createAllBadOpcodeTests.sh (hunk 5 above)
Comment (qbzzt) [reply src/Templates/DiffPlaces/createAllBadOpcodeTests.sh]: Yes, they are used for opcodes that have an immediate operand. See https://hackmd.io/@shemnon/CancunOpcodeAllocation .

File: src/Templates/DiffPlaces/templateGen.js (hunk 6 above)
Comment (qbzzt) [reply src/Templates/DiffPlaces/templateGen.js]: Ask and [you shall receive](https://github.com/ethereum/tests/blob/9979ea5d63f1e43221e74f7f7b255f415655aaa3/src/Templates/DiffPlaces/createDiffPlacesTests.sh#L8).

File: src/GeneralStateTestsFiller/stBadOpcode/invalidDiffPlacesFiller.yml [hunk 7]
```diff
@@ -678,7 +678,11 @@ invalidDiffPlaces:
         value: !!int -1
 
       network:
-        - '>=London'
+        - 'London'
+        - 'Merge'
+        - 'Shanghai'
+        - 'Cancun'
```
Comment (winsvega) [inline src/GeneralStateTestsFiller/stBadOpcode/invalidDiffPlacesFiller.yml:684]: why is it fixed now?

File: src/GeneralStateTestsFiller/stSelfBalance/diffPlacesFiller.yml [hunk 8]
```diff
@@ -650,7 +650,11 @@ diffPlaces:
         value: !!int -1
 
       network:
-        - '>=London'
+        - 'London'
+        - 'Merge'
+        - 'Shanghai'
+        - 'Cancun'
```
Comment (winsvega) [inline src/GeneralStateTestsFiller/stSelfBalance/diffPlacesFiller.yml:656]: `>= Cancun` ???

**Pull request merged**